x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/CertificateToolTests.java

```diff
@@ -892,6 +891,44 @@ Collection<CertificateInformation> getCertificateInformationList(Terminal termin
         assertThat("For command line option " + optionThatTriggersZip, isZip.get(), equalTo(true));
     }
 
+    public void testErrorIfSigningCertificateAndKeyDontMatch() throws Exception {
```

Comment:
This test fails for me when a generated password is larger than 50 chars:
https://gradle-enterprise.elastic.co/s/7qquwban7ymjw

Reproduce with:
```
./gradlew ':x-pack:plugin:security:cli:test' --tests "org.elasticsearch.xpack.security.cli.CertificateToolTests.testErrorIfSigningCertificateAndKeyDontMatch" -Dtests.seed=8DE86CB48F6EC3DF -Dtests.locale=es-CL -Dtests.timezone=Etc/GMT-4 -Druntime.java=21
```

Reply:
I took a closer look and the problem is with calling `generateCA` method twice. It fails in case when the first call creates a password larger than 50 chars and the second does not. The terminal will contain the output of both commands and assertion in the second call will fail. The easiest solution is to simply call `terminal.reset();` after first `generateCA` call.